test/other/test_mmap_empty.c

```diff
@@ -0,0 +1,35 @@
+#include <assert.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <limits.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/mman.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+int main() {
+  int fd = open("foo", O_RDWR|O_CREAT); // create an empty file
+  assert(fd >= 0);
+
+  struct stat st;
+  assert(fstat(fd, &st) != -1);
+  int filesize = st.st_size;
+  assert(filesize == 0);
+
+  // mmap of an empty file should fail with EINVAL
+  char *mapped = (char*) mmap(NULL, 0, PROT_READ, MAP_PRIVATE, fd, 0);
+  assert(mapped == MAP_FAILED);
+  printf("mmap failed with: %s\n", strerror(errno));
+  assert(errno == EINVAL);
+
+  // same if we ask for PAGE_SIZE bytes
+  mapped = (char*) mmap(NULL, PAGE_SIZE, PROT_READ, MAP_PRIVATE, fd, 0);
+  assert(mapped == MAP_FAILED);
+  printf("mmap failed with: %s\n", strerror(errno));
+  assert(errno == EINVAL);
```

Comment:
Should we also check the flip case, of the file *not* being empty, but we ask for 0 bytes?

Reply:
Done.

Also tested and fixed under wasmfs